src/main/java/com/hivemq/cli/mqtt/MqttClientExecutor.java

```diff
@@ -174,20 +198,25 @@ void mqtt5Publish(
 
         final Mqtt5Publish publishMessage = publishBuilder.build();
 
-        Logger.debug("{} sending PUBLISH ('{}') {}",
+        Logger.debug("{} sending PUBLISH ('{}')\n    {}",
                 clientLogPrefix,
                 bufferToString(publishOptions.getMessage()),
                 publishMessage);
 
         client.toAsync().publish(publishMessage).whenComplete((publishResult, throwable) -> {
             if (throwable != null) {
-                Logger.error(throwable,
-                        "{} failed PUBLISH to TOPIC '{}': {}",
+                Logger.error("{} failed PUBLISH to TOPIC '{}': {}",
                         clientLogPrefix,
                         topic,
                         Throwables.getRootCause(throwable).getMessage());
+                if (throwable instanceof Mqtt5PubAckException) {
+                    Logger.debug(((Mqtt5PubAckException) throwable).getMqttMessage());
+                } else if (throwable instanceof Mqtt5PubRecException) {
+                    Logger.debug(((Mqtt5PubRecException) throwable).getMqttMessage());
+                }
+                Logger.trace(throwable);
```

Comment:
This is very spammy.

Reply:
I rethought the logging and changed it to be a better fit.